homeassistant/components/coinbase.py

```diff
@@ -0,0 +1,65 @@
+"""
+Support for Coinbase.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/coinbase/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_API_KEY
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['coinbase==2.0.6']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'coinbase'
+
+CONF_API_SECRET = 'api_secret'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=5)
+
+NETWORK = None
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Required(CONF_API_SECRET): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the Coinbase component.
+
+    Will automatically setup sensors to support
+    wallets discovered on the network.
+    """
+    # pylint: disable=global-statement, import-error
+    global NETWORK
+    api_key = config[DOMAIN].get(CONF_API_KEY)
+    api_secret = config[DOMAIN].get(CONF_API_SECRET)
+
+    NETWORK = CoinbaseData(api_key, api_secret)
+
+    return True
+
+
+class CoinbaseData(object):
+    """Get the latest data and update the states."""
+
+    def __init__(self, api_key, api_secret):
+        """Init the Ecobee data object."""
```

Comment:
Oh cool. Thermostats on the blockchain? ;-) (Honestly, 90% of suggested blockchain applications are probably less useful than 'thermostat on blockchain', but I digress.)

Reply:
lol 😆